### ISSUE TITLE
Allow to copy an existing mock

### ISSUE THREAD

**Issue by polarene** [mockk/mockk#1213]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

I've encountered a scenario where I needed more than one mock of the same type but with a subset of different stubbings. Initializing each mock from scratch leads to duplication, which can only be avoided writing a custom function. 
I'd like to have a copy factory function so I could write:
```
val first = mockk<Car> {
  // base stubbings
  every { drive(Direction.NORTH) } returns Outcome.OK
  every { door(DoorType.FRONT_LEFT).windowState() } returns WindowState.UP
}
val second = mockk(first) {
  // redefine a subset of stubbings
  every { drive(Direction.SOUTH) } returns Outcome.ERROR
}
```
`second` will be a new mock instance having all the configuration copied over, and its stubbings don't affect `first`.

### Current Behavior

The library currently doesn't support this feature.

**Comment by polarene** [comment]
I've started working on this feature but I'm still studying how to clone the internal data structures.

**Comment by PandolfiJoaquin** [comment]
I could help contributing to this feature. It would help me a lot to have this

**Comment by Raibaz** [comment]
Sure, go ahead!

**Comment by polarene** [comment]
Hi @PandolfiJoaquin, thanks for joining! Unfortunately, I abandoned the research because I couldn't find a way to clone a proxy from a target type reference. I think the easiest solution lies at the test level: just extract an extension function with your desired stubbings so you can apply them to several mocks. Taking the example in this issue:
```kotlin
val baseStubbings: Car.() -> Unit = {
    every { drive(Direction.NORTH) } returns Outcome.OK
    every { door(DoorType.FRONT_LEFT).windowState() } returns WindowState.UP
}

val first = mockk<Car>(block = baseStubbings)
val second = mockk<Car> {
    baseStubbings()
    // redefine a subset of stubbings
    every { drive(Direction.SOUTH) } returns Outcome.ERROR
}
```

**Comment by polarene** [comment]
We should probably update the documentation promoting this pattern. I discovered it at work, where we had to repeat many verifications, so we found out it was possible by extracting an extension function on `MockKVerificationScope`. 
You can do the same on `MockKMatcherScope`, for example, to extract common argument matchers.

**Comment by PandolfiJoaquin** [comment]
This does seem to do the trick. Thank you!

**Comment by Raibaz** [comment]
Thanks for the workarund, @polarene !

Would either you or @PandolfiJoaquin be able to send out a PR to document this?

**Comment by PandolfiJoaquin** [comment]
@Raibaz Is there any place in the documentation where this kind of patterns are documented? It feels a out of place to put it in the [features section](https://github.com/mockk/mockk?tab=readme-ov-file#features) but most guides and recipes are links to external sites

**Comment by polarene** [comment]
Thanks, happy it helped! I will gladly contribute to the documentation, maybe we can create a new section like "usage patterns", "common patterns", etc.

**Comment by Raibaz** [comment]
Good idea! I like it! It would also be a good place to document that people shouldn't mock data classes 😁

**Comment by polarene** [comment]
Totally agree, let's document both patterns and anti-patterns!
